### PR TITLE
Add support for graphql-schema-linter

### PR DESCRIPTION
**Pull request by steverice**
This adds Arcanist linter support for [graphql-schema-linter](https://github.com/cjoudrey/graphql-schema-linter) and all of its configuration options.

As this brings the total number of Node-based linters up to 4, extract common functionality used by all of them into `NodeExternalLinter` as done in 89815d14eb67e249b38f87dcd5db768611964c3f.

### CODE REVIEW

**Review by jparise — APPROVED** [review]
Looks good!

File: src/GraphQLSchemaLinter.php [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            ->setName($this->getLinterName())
+            ->setLine($error['location']['line'])
+            ->setChar($error['location']['column'])
+            ->setDescription($error['message'])
+            ->setSeverity($this->getDefaultMessageSeverity($rule));
+          $messages[] = $message;
+        }
+        break;
+      case 2: // If the process exits with 2 it means an invalid configuration was provided.
+        break;
```
Comment (jparise) [inline src/GraphQLSchemaLinter.php:194]: Should we print something in this case?

File: src/ESLintLinter.php [hunk 2]
```diff
@@ -166,11 +128,11 @@ protected function parseLinterOutput($path, $err, $stdout, $stderr) {
         if (strpos($offense['message'], "File ignored ") === 0) {
           continue;
         }
-        
+
         $message = new ArcanistLintMessage();
         $message->setPath($file['filePath']);
         $message->setSeverity($this->mapSeverity($offense['severity']));
-        $message->setName(idx($offense, 'ruleId', 'unknown'));
+        $message->setName(idx($offense, 'ruleId', 'unknown') || 'unknown');
```
[the remark this answers is not in the document]
Comment (steverice) [reply src/ESLintLinter.php:135]: Ah yes, that's a bugfix for this linter. But I didn't mean to include it here. I'll remove.

File: src/GraphQLSchemaLinter.php (hunk 1 above)
Comment (steverice) [reply src/GraphQLSchemaLinter.php:194]: Yes — I wasn't able to reproduce this case but I will try again and see if I can print something reasonable.

**Review by jparise — APPROVED** [review]

**Pull request merged**